cpp/NamingConventions.md

```diff
@@ -0,0 +1,192 @@
+# BlueBrain HPC Team C++ Naming Conventions
+
+This document describes how C++ symbols should be named.
+
+## Common Conventions
+
+### Use descriptive names
+
+```cpp
+// no
+int bufsz;
+
+// yes
+int buffer_size;
+```
+
+Generally use `snake_case` style as it is compliant with spell checkers.
+
+#### Exception
+
+Function parameters and local variables used only within less than 3 lines can break this rule, for instance:
```

Comment:
```suggestion
Function parameters and local variables used only within the next 3 lines can break this rule, for instance:
```
I find `within less than` confusing, since `within` already containing a `less than the limit` sort of. I would personally raise that limit a little, too, but oh well.